transports/ssl/src/main/java/org/mule/transport/ssl/DefaultTlsContextFactory.java

```diff
@@ -28,18 +35,76 @@
 
 /**
  * Default implementation of the {@code TlsContextFactory} interface, which delegates all its operations to a
- * {@code TlsConfiguration} object.
+ * {@code TlsConfiguration} object. Only enabled cipher suites and protocols will not delegate to it if configured.
  */
-public class DefaultTlsContextFactory implements TlsContextFactory
+public class DefaultTlsContextFactory implements TlsContextFactory, Initialisable
 {
+
     private static final Logger logger = LoggerFactory.getLogger(DefaultTlsContextFactory.class);
+    private static final String DEFAULT = "default";
 
     private String name;
 
     private TlsConfiguration tlsConfiguration = new TlsConfiguration(null);
 
-    private boolean initialized = false;
+    private AtomicBoolean initialized = new AtomicBoolean(false);
     private boolean trustStoreInsecure = false;
+    private String[] enabledProtocols;
+    private String[] enabledCipherSuites;
+
+    @Override
+    public void initialise() throws InitialisationException
+    {
+        if (initialized.getAndSet(true))
+        {
+            return;
+        }
+
+        try
+        {
+            tlsConfiguration.initialise(null == getKeyStorePath(), null);
+        }
+        catch (CreateException e)
+        {
+            throw new InitialisationException(MessageFactory.createStaticMessage("Unable to initialise TLS configuration"), e, this);
+        }
+
+        if (!isUseDefaults(enabledProtocols))
+        {
+            String[] globalEnabledProtocols = tlsConfiguration.getEnabledProtocols();
+            if (globalEnabledProtocols != null)
+            {
+                String[] validProtocols = ArrayUtils.intersection(enabledProtocols, globalEnabledProtocols);
+                if (validProtocols.length < enabledProtocols.length)
+                {
+                    globalConfigNotHonored("protocols", globalEnabledProtocols);
+                }
+            }
+        }
+
+        if (!isUseDefaults(enabledCipherSuites))
+        {
+            String[] globalEnabledCipherSuites = tlsConfiguration.getEnabledCipherSuites();
+            if (globalEnabledCipherSuites != null)
+            {
+                String[] validCipherSuites = ArrayUtils.intersection(enabledCipherSuites, globalEnabledCipherSuites);
+                if (validCipherSuites.length < enabledCipherSuites.length)
+                {
+                    globalConfigNotHonored("cipher suites", globalEnabledCipherSuites);
+                }
+            }
+        }
+    }
+
+    private boolean isUseDefaults(String[] array)
+    {
+        return (array == null) || ((array.length == 1) && DEFAULT.equalsIgnoreCase(array[0]));
```

Comment:
i'll merge it but unnecessary parenthesis makes this harder to read than it should